crates/sound/src/device.rs

```diff
@@ -472,37 +470,34 @@ impl VhostUserSoundThread {
                             u32::from(
                                 self.streams.read().unwrap()[stream_id as usize]
                                     .params
-                                    .buffer_bytes,
+                                    .period_bytes,
                             ) as usize,
                             descriptor.len(),
                         )
                         .into());
                     }
-                    IoState::WaitingBufferForStreamId(_stream_id) => {
-                        buffers.push(Buffer::new(*descriptor, Arc::clone(&message)));
+                    IoState::WaitingBufferForStreamId(_) => {
+                        buffers.push(Buffer::new(*descriptor, Arc::clone(&message), !write));
```

Comment:
One point more for the enum.
I'd suggest to uniform `write/read`, for example here we use `write` and `is_rx` in Buffer. It is a bit confusing IMHO.

Reply:
+1

Reply:
Changed with:

> Added a Direction enum